blocks/HARDWARE/SIGNAL_ANALYZERS/TEKTRONIX/RSA500/BLOCK_IQ_RSA500/BLOCK_IQ_RSA500.py

```diff
@@ -0,0 +1,118 @@
+from flojoy import flojoy, DataContainer, OrderedPair
+from typing import Optional, TypedDict
+from flojoy.instruments.tektronix.RSA_API import *  # noqa: F403
+from ctypes import cdll, c_int, c_bool, c_double, c_float, byref
+from os import path
+import numpy as np
+
+
+class IQSplitOutput(TypedDict):
+    I_real: OrderedPair
+    Q_imag: OrderedPair
+
+
+@flojoy
+def BLOCK_IQ_RSA500(
+    input: Optional[DataContainer] = None,
+    center_freq: float = 100e6,
+    ref_level: float = -30,
+    record_length: float = 1e3,
+    bandwidth: float = 5e5,
+) -> IQSplitOutput:
+    """Extract Block IQ measurement from a Tektronix RSA.
+
+    This block should also work with compatible Tektronix RSAXXX instruments.
+
+    Tested with RSA507a.
+
+    Parameters
+    ----------
+    center_freq : float, default=100e6
+        The center frequency, in Hz.
+    ref_level : float, default=-30
+        The reference level (the maximum y axis value), in dBm.
+    record_length : float, default=1e3
+        The length to record (length of x axis), in ms.
+    bandwidth : float, default=5e5
+        Resolution bandwidth, in Hz.
+
+    Returns
+    -------
+    I_real: OrderedPair
+        x: time
+        y: power
+    Q_imag: OrderedPair
+        x: time
+        y: power
+    """
+
+    # Connect to RSA
+    defau = "C:/Tektronix/RSA_API/lib/x64/RSA_API.dll"
+    moved = "C:/Program Files/Tek/RSA_API/lib/x64/RSA_API.dll"
+    if path.isfile(defau):
+        rsa = cdll.LoadLibrary(defau)
+    elif path.isfile(moved):
+        rsa = cdll.LoadLibrary(moved)
+    else:
+        raise FileNotFoundError(
+            "Cannot find RSA_API.dll. Download from: https://www.tek.com/en/products/spectrum-analyzers/rsa500"
+        )
```

Comment:
Is this dll always located here? Also, this only works on windows?

Reply:
Yeah it's only available for Windows. The default install location is the second one but I moved mine because I didn't want it on C: directly.